src/sentry/api/endpoints/project_details.py

```diff
@@ -647,17 +651,78 @@ def put(self, request: Request, project) -> Response:
         if "dynamicSampling" in result:
             raw_dynamic_sampling = result["dynamicSampling"]
             fixed_rules = self._fix_rule_ids(project, raw_dynamic_sampling)
+
             if project.update_option("sentry:dynamic_sampling", fixed_rules):
+
                 changed_proj_settings["sentry:dynamic_sampling"] = result["dynamicSampling"]
 
+                if old_data.get("dynamicSampling") is not None:
+                    old_rules = old_data["dynamicSampling"].get("rules") or []
+                    rules = result["dynamicSampling"].get("rules") or []
+
+                    if len(rules) > len(old_rules):
+                        self.create_audit_entry(
+                            request=request,
+                            organization=project.organization,
+                            target_object=project.id,
+                            event=audit_log.get_event_id("SAMPLING_RULE_ADD"),
+                            data=project.get_audit_log_data(),
+                        )
+                        data = serialize(project, request.user, DetailedProjectSerializer())
+                        return Response(data)
+
+                    if len(rules) < len(old_rules):
+                        self.create_audit_entry(
+                            request=request,
+                            organization=project.organization,
+                            target_object=project.id,
+                            event=audit_log.get_event_id("SAMPLING_RULE_REMOVE"),
+                            data=project.get_audit_log_data(),
+                        )
+                        data = serialize(project, request.user, DetailedProjectSerializer())
+                        return Response(data)
+
+                    for index, rule in enumerate(rules):
+                        if rule["active"] != old_rules[index]["active"]:
+                            if rule["active"] is True:
+                                self.create_audit_entry(
+                                    request=request,
+                                    organization=project.organization,
+                                    target_object=project.id,
+                                    event=audit_log.get_event_id("SAMPLING_RULE_ACTIVATE"),
+                                    data=project.get_audit_log_data(),
+                                )
+                                data = serialize(project, request.user, DetailedProjectSerializer())
+                                return Response(data)
```

Comment:
These early returns will result in request that contain both `dynamicSampling` and other request attributes to only be partially applied. Is that intentional?

Reply:
yes! when adding, editing, and removing dynamic sampling configurations, the requests only contain dynamic sampling changes so this shall be fine.

Reply:
Do we have any logic that lets users know that making a request with both dynamicSampling and other attributes will not work?

Reply:
I don't see how they can do it via our App...are you saying that they might want to send different configurations somewhere else? is this API public?

Reply:
> is this API public?

[It is](https://docs.sentry.io/api/projects/update-a-project/) but not all the body parameters are documented.

Reply:
oh I see

Reply:
That extra audit log helps cover the scenarios where we get mixed payloads though.

Reply:
I have updated the code and it shall be fine now 😉